src/ansys/fluent/core/quantity.py

```diff
@@ -1,1369 +1,869 @@
+                    unit_str=self._units_table.derived_units[unit_term]["composition"],
+                    power=unit_term_power,
+                    dimensions=dimensions,
+                )
 
-    def _get_converted_temperature(self, to_unit_str):
-        """Get converted temperature."""
-        from_unit_str = self.unit
+        return dimensions
 
-        if "^" in from_unit_str:
-            from_unit_str = from_unit_str.split("^")[0]
+    @property
+    def unit_str(self):
+        """Unit string representation of dimensions"""
+        return self._unit_str
 
-        if "^" in to_unit_str:
-            to_unit_str = to_unit_str.split("^")[0]
+    @property
+    def dimensions(self):
+        """Dimensions representation of unit string"""
+        return self._dimensions
 
-        _, prefix = filter_multiplier(
-            from_unit_str, lambda item, unit: len(unit) > 0.0 and unit.startswith(item)
-        )
-        from_unit_multiplier = (
-            _UnitsTable.multipliers[prefix] if len(prefix) != 0 else 1.0
-        )
 
-        _, prefix = filter_multiplier(
-            to_unit_str, lambda item, unit: len(unit) > 0.0 and unit.startswith(item)
-        )
-        to_unit_multiplier = (
-            _UnitsTable.multipliers[prefix] if len(prefix) != 0 else 1.0
-        )
+class QuantityMap(object):
+    """Creates a Quantity Map object based on a given quantity map.
 
-        from_key = remove_multiplier(from_unit_str)
-        to_key = remove_multiplier(to_unit_str)
+    Parameters
+    ----------
+    quantity_map : dict
+        Dictionary containing quantity map units and values.
 
-        return Quantity(
-            _UnitsTable.temperature_conversions[from_key][to_key](
-                from_unit_multiplier * self.value
-            )
-            / to_unit_multiplier,
-            to_unit_str,
-        )
+    Properties
+    ----------
+    unit_str : str
+        Unit string representation of quantity map.
 
-    def _get_si_unit(self, other, func):
-        """Get SI unit."""
-        curr_dim = self.get_dimensions_list()
-        other_dim = other.get_dimensions_list()
-        temp_dim = [func(curr_dim[i], other_dim[i]) for i in range(len(other_dim))]
-        temp_unit = get_si_unit_from_dim(temp_dim)
-        return temp_unit
+    Returns
+    -------
+    QuantityMap instance.
+    """
 
-    def __pow__(self, exponent):
-        new_dims = list(map(lambda x: x * exponent, self.get_dimensions_list()))
-        new_si_unit = get_si_unit_from_dim(new_dims)
-        new_si_value = pow(self._si_value, exponent)
-        return Quantity(new_si_value, new_si_unit)
+    def __init__(self, quantity_map):
+        self._units_table = UnitsTable()
+        self._unit_str = self._map_to_unit_str(quantity_map)
 
-    def __str__(self):
-        return f'({self.value}, "{self.unit}")'
+    def _map_to_unit_str(self, quantity_map: dict) -> str:
+        """Convert a quantity map into a unit string.
```

Comment:
I made docstrings for all functions and classes so that it's easier for developers to understand what a function or class does in the event that the code needs to be refactored in the future. If that's not required then I can go through and remove the docstrings on private members.

For `map_to_unit_str` and `unit_str_to_dim` I thought these should be private because they're only used by the quantity class and would not be called directly. Within the tests, getting a unit string from a quantity map look like this: 

```
v = q.Quantity(40.2, quantity_map={"Velocity": 1.0})
v.units
>>> "m s^-1"
``` 

and getting dimensions from a unit string would look like this:

```
v = q.Quantity(20.2, units="m s^-1")
v.dimensions
>>> [0.0, 1.0, -1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
```

Reply:
My only concern was if this was documentation that you actually wanted to show up in the online documentation. If this was specifically for developers, then this is fine by me.

Regarding whether to make these private or not, if users are not expected to ever use it directly, then making it private also makes sense to me, thanks for the explanation.